docs/modelserving/v1beta1/llm/huggingface/multi-node/README.md

```diff
@@ -0,0 +1,315 @@
+# Multi-node/Multi-GPU Inference with Hugging Face LLM Serving Runtime
+
+This guide provides step-by-step instructions on setting up multi-node and multi-GPU inference using Hugging Face's LLM Serving Runtime. Before proceeding, please ensure you meet the following prerequisites and understand the limitations of this setup.
+
+## Prerequisites
+
+- Multi-node functionality is only supported in **RawDeployment** mode.
+- **Auto-scaling is not available** for multi-node setups. The auto-scaler will be automatically set to external.
+- A **Persistent Volume Claim (PVC)** is required for multi-node configurations, and it must support the **ReadWriteMany (RWM)** access mode.
+
+
+### Key Validations
+
+- `TENSOR_PARALLEL_SIZE` and `PIPELINE_PARALLEL_SIZE` cannot be set via environment variables. They must be configured through `workerSpec.tensorParallelSize` and `workerSpec.pipelineParallelSize`.
+- In a ServingRuntime designed for multi-node, both `workerSpec.tensorParallelSize` and `workerSpec.pipelineParallelSize` must be set.
+- The minimum value for `workerSpec.tensorParallelSize` is **1**, and the minimum value for `workerSpec.pipelineParallelSize` is **2**.
+- Currently, four GPU types are allowed: `nvidia.com/gpu` (*default*), `intel.com/gpu`, `amd.com/gpu`, and `habana.ai/gaudi`.
+- You can specify the GPU type via InferenceService, but if it differs from what is set in the ServingRuntime, both GPU types will be assigned to the resource. Then it can cause issues.
+- The Autoscaler must be configured as `external`.
+- The only supported storage protocol for StorageURI is `PVC`.
+
+!!! note 
+
+    You must have **exactly one head pod** in your setup. The replica count for this head pod can be adjusted using the `min_replicas` or `max_replicas` settings in the `InferenceService (ISVC)`. However, creating additional head pods will cause them to be excluded from the Ray cluster, resulting in improper functioning. Ensure this limitation is clearly documented.
```

Comment:
is there a way to limit this in the code?

Reply:
This is kind of generic features so adding checking logic would not be simple.